setup.py

```diff
@@ -317,7 +317,16 @@ def try_strip_symbols(so_file):
 
     def build_extension(self, ext):
         if isinstance(ext, CMakeExtension):
-            self.build_extension_cmake(ext)
+            try:
+                self.build_extension_cmake(ext)
+            except subprocess.CalledProcessError as e:
+                print("WARNING: Command '{}' returned non-zero exit status {}.".format(e.cmd, e.returncode))
+                if not ext.optional:
+                    raise
+            except Exception as e:
+                print("WARNING: An error occurred while building the CMake extension.")
+                if not ext.optional:
+                    raise
```

Comment:
If I understand correctly, we can skip the build error, but then we call:

```python
self.try_strip_symbols(self.get_ext_fullpath(ext.name))
```

and if the file wasn't built correctly, we'll get a new error:
```
WARNING: Command '['cmake', '-S/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/ddtrace/appsec/_iast/_taint_tracking', '-B/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/build/cmake.linux-x86_64-cpython-39', '-DPYTHON_EXECUTABLE=/usr/local/bin/python3', '-DCMAKE_BUILD_TYPE=Release', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/build/lib.linux-x86_64-cpython-39/ddtrace/appsec/_iast/_taint_tracking', '-DLIB_INSTALL_DIR=/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/build/lib.linux-x86_64-cpython-39/ddtrace/appsec/_iast/_taint_tracking', '-DEXTENSION_NAME=_native.cpython-39-x86_64-linux-gnu.so']' returned non-zero exit status 1.
2511  strip: 'build/lib.linux-x86_64-cpython-39/ddtrace/appsec/_iast/_taint_tracking/_native.cpython-39-x86_64-linux-gnu.so': No such file <!----
```

Reply:
Absolutely, good catch.

Reply:
Made this more permissive, thank you for the report!